### DISCUSSION TITLE
Flashing step fails (Roborock S7)

### DISCUSSION THREAD

**Discussion by Californian** [Hypfer/Valetudo#1795] [Q&A / Support] [answered]
Hello,

Love the project, very impressed with all the work that's going into it!

I have successfully disassembled my roborock S7 and connected TPA17 to SH1 and get the flashing mode usb device (`lsusb` shows `Bus 001 Device 034: ID 1f3a:efe8 Allwinner Technology sunxi SoC OTG connector in FEL/flashing mode`).

However, when I run `./run.sh` (as root), I just receive `ERROR: Allwinner USB FEL device not found!` a bunch of times (with a wait message after the second). (I'll note here that it might be useful to add `set -e` after the first "shebang" (`#!`) line so that the script exits after any commands fail, but that's very minor).

I tried just running some diagnostic `sunxi-fel` commands, but to no avail.

Specifically, I ran and got the following:
```sh
[root@comp]# sunxi-fel -v -d 001:034
Selecting USB Bus 001 Device 034
libusb_open() ERROR -1: Input/Output Error
```

Very interestingly, after running that command, the device number in `lsusb` changes.

I tried googling, but nothing relevant comes up.

Any ideas how to debug? I'm quite experienced with linux but haven't done much hardware hacking.

**Comment by Hypfer** [comment, marked as answer]
This can happen under some circumstances with specific combinations of hardware and software.

Things that you can try to fix this:
- Use a different USB Cable
- Use a different USB Port
- Use a different USB Controller
- Use a different Host altogether
- Use a different operating system (a live usb stick should be fine)

**Reply by Californian** [reply]
Thanks! I used a different computer and that worked, all up and running now!
